### DISCUSSION TITLE
Is there any way to define margins inside the canvas?

### DISCUSSION THREAD

**Discussion by Shilpa-K07** [fabricjs/fabric.js#7144] [Q&A] [answered]
**Can we define margins inside canvas, so that object or part of object which goes out of that defined margin should be hidden.**

**Comment by melchiar** [comment, marked as answer]
see http://fabricjs.com/clippath-part3

**Reply by Shilpa-K07** [reply]
Thanks It worked well. Can't we give border or stroke color to this clip path?

**Reply by melchiar** [reply]
Not natively no, clipPaths are for clipping only

**Comment by asturur** [comment]
What you can do is an overlay.
You can creat an overlay that is either fully opaque or semi transparent, that is drawn on top of objects at the end of rendering.
This will not affect controls.

**Comment by Shilpa-K07** [comment]
Facing one more issue with object centering inside clipPath..Is there any built-in methods available?
